napari/layers/shapes/shapes.py

```diff
@@ -2141,6 +2134,14 @@ def expand_shape(self, data):
         data_full : array
             Full D dimensional data array of the shape.
         """
+        warnings.warn(
+            "expand_shape is deprecated and will be removed in version 0.4.8."
+            " It should no longer be used as layers should will soon not know"
+            " which dimensions are displayed.",
```

Comment:
As before, I don't love removing functionality without a stopgap for how to do it in the future... Do you have an idea of how we will do this? Perhaps rather than deprecating it altogether, add a required argument of `to_dims` or similar?

Reply:
This really is an internal method that I don't expect a user to have ever used, and now we don't need it. It should never have been public, but I've also added some info about trying to stay using the full nD representation the whole time. See also my comment here https://github.com/napari/napari/pull/2327/files#r586552820